Src/WitsmlExplorer.Api/Workers/BaseWorker.cs

```diff
@@ -59,6 +67,56 @@ protected string CancellationReason()
         {
             return "The job was cancelled by the user.";
         }
+        protected async Task<WitsmlCapServers> GetTargetServerCapabilities()
+        {
+            if (_targetServerCapabilities == null)
+            {
+                _targetServerCapabilities = await GetTargetWitsmlClientOrThrow().GetCap();
+            }
+            return _targetServerCapabilities;
+        }
+
+        protected List<WitsmlLogs> GetUpdateLogDataQueries(WitsmlLog log, WitsmlLogData logData, int chunkSize, string mnemonicList)
```

Comment:
You only use the WitsmlLog to get the uids. At some of the places where you use this function, you create a new WitsmlLog only to add the uids. In this case it might be better to accept the uids separately instead.

Reply:
done